model/versioning/Versioned.php

```diff
@@ -634,87 +634,75 @@ public function augmentDatabase() {
-
-						foreach($versionedTables as $child) {
-							if($table === $child) break; // only need subclasses
-
-							// Select all orphaned version records
-							$orphanedQuery = SQLSelect::create()
-								->selectField("\"{$table}_versions\".\"ID\"")
-								->setFrom("\"{$table}_versions\"");
-
-							// If we have a parent table limit orphaned records
-							// to only those that exist in this
-							if(DB::get_schema()->hasTable("{$child}_versions")) {
-								$orphanedQuery
-									->addLeftJoin(
-										"{$child}_versions",
-										"\"{$child}_versions\".\"RecordID\" = \"{$table}_versions\".\"RecordID\"
-										AND \"{$child}_versions\".\"Version\" = \"{$table}_versions\".\"Version\""
-									)
-									->addWhere("\"{$child}_versions\".\"ID\" IS NULL");
-							}
-
-							$count = $orphanedQuery->count();
-							if($count > 0) {
-								DB::alteration_message("Removing {$count} orphaned versioned records", "deleted");
-								$ids = $orphanedQuery->execute()->column();
-								foreach($ids as $id) {
-									DB::prepared_query(
-										"DELETE FROM \"{$table}_versions\" WHERE \"ID\" = ?",
-										array($id)
-									);
-								}
-							}
-						}
-					}
-				}
+				// Cleanup any orphans
+				$this->cleanupVersionedOrphans("{$suffixBaseTable}_versions", "{$suffixTable}_versions");
 
-				DB::require_table("{$table}_versions", $versionFields, $versionIndexes, true, $options);
+				// Build versions table
+				DB::require_table("{$suffixTable}_versions", $versionFields, $versionIndexes, true, $options);
 			} else {
-				DB::dont_require_table("{$table}_versions");
+				DB::dont_require_table("{$suffixTable}_versions");
 				if($this->hasStages()) {
-					$liveTable = $this->stageTable($table, static::LIVE);
+					$liveTable = $this->stageTable($suffixTable, static::LIVE);
 					DB::dont_require_table($liveTable);
 				}
 			}
 		}
 	}
 
 	/**
+	 * Cleanup orphaned records in the _versions table
+	 *
+	 * @param string $baseTable base table to use as authoritative source of records
+	 * @param string $childTable Sub-table to clean orphans from
+	 */
+	protected function cleanupVersionedOrphans($baseTable, $childTable) {
+		// Skip if child table doesn't exist
+		if(!DB::get_schema()->hasTable($childTable)) {
+			return;
+		}
+		// Skip if tables are the same
+		if($childTable === $baseTable) {
+			return;
+		}
+
+		// Select all orphaned version records
+		$orphanedQuery = SQLSelect::create()
+			->selectField("\"{$childTable}\".\"ID\"")
+			->setFrom("\"{$childTable}\"");
+
+		// If we have a parent table limit orphaned records
+		// to only those that exist in this
+		if(DB::get_schema()->hasTable($baseTable)) {
+			$orphanedQuery
+				->addLeftJoin(
+					$baseTable,
+					"\"{$childTable}\".\"RecordID\" = \"{$baseTable}\".\"RecordID\"
+					AND \"{$childTable}\".\"Version\" = \"{$baseTable}\".\"Version\""
+				)
+				->addWhere("\"{$baseTable}\".\"ID\" IS NULL");
```

Comment:
> escaped quotes = just as much ewww

It's the standard for now, and I don't think we can get away from it. I made a failed attempted and realised it was much more practical to keep the convention than try to encapsulate it.